scheduler/src/cook/mesos/scheduler.clj

```diff
@@ -526,10 +550,42 @@
      :backfill-jobs (mapv :job/uuid jobs-to-backfill)
      :matched-head? (matched? head-of-considerable)}))
 
+(defn task->mesos-message
+  [t]
+  (let [cmd (:command t)
+        resources (:resources t)]
+    {:name (:name t)
+     :task-id [{:value (:task-id t)}]
+     :slave-id [{:value (:slave-id t)}]
+     :resources (vec (remove nil?
+                             [{:type :value-scalar,
+                               :name "mem",
+                               :scalar (:mem resources)
+                               :role (:role t)}
```

Comment:
I'm concerned this will not work in one of two cases.

1. Cook is started with role "cook" but gets offers with resources with role "*"
2. Cook is started with role "cook" but gets offers with resources with role "*" and role "cook"

I make a special case for the last one because it will require being a bit clever about how the resources are mapped. For example, imagine the job will fit if you add the "*" and "cook" resources but would not otherwise. 

I expect what you will end up having to do is take in the matches of offers and tasks such that you can parcel out correctly.